### ISSUE TITLE
1.0.3-debian12 -  CTRL-T/CTRL-SHIFT-F locks the xfce window/display manager

### ISSUE THREAD

**Issue by keypey** [giuspen/cherrytree#2398]
**Version, Operative system**
```
Distributor ID: Debian
Description:    Debian GNU/Linux 12 (bookworm)
Release:    12
Codename:   bookworm
Linux 6.1.0-15-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.1.66-1 (2023-12-09) x86_64 GNU/Linux
Display manager: LightDM
Desktop: XFCE
```

**Describe the bug**
Searching for headings (sometimes the first press): CTRL-T locks the window/display manager
All window frames disappear. 
The system stops responding, all windows of all workspaces and separate monitors appear on the primary screen.

**To Reproduce**
- Open a file (SQLite) in which we have many branches
- press CTRL-T/CTRL-SHIFT-F, enter string, press ENTER
- I suspect that during operations opening the window with search results

**Screenshots**
I can't take a screenshot after pressing CTRL-T.

**Workaround**
Install the version cherrytree_1.0.2-2~Debian12_amd64.deb
In my case, in order not to lose data of other open applications, I do:
- CTRL-ALT-F1 - kill cherrytree process.
- CTRL-ALT-F7 returns to the graphical environment.
- From the terminal (run possible if we have a keyboard shortcut to launch the terminal):
xfce4-panel -r
- Termination of other programs
- Restart the system.

**Additional information**
Along with installing the new version (1.0.3) of cherrytree, I updated the packages.
I don't suspect it will have an impact - because version 1.0.2 works correctly - but I've attached a list of updated packages.
```log
Start-Date: 2023-12-12  07:45:17
Commandline: apt full-upgrade -y
Requested-By: bruno (1000)
Install: linux-headers-6.1.0-15-common:amd64 (6.1.66-1, automatic), linux-headers-6.1.0-15-amd64:amd64 (6.1.66-1, automatic), linux-image-6.1.0-15-amd64:amd64 (6.1.66-1, automatic)
Upgrade: 
        libreoffice-l10n-en-gb:amd64 (4:7.4.7-1, 4:7.4.7-1+deb12u1), 
        libreoffice-l10n-en-za:amd64 (4:7.4.7-1, 4:7.4.7-1+deb12u1), 
        libreoffice-calc:amd64 (4:7.4.7-1, 4:7.4.7-1+deb12u1), 
        libreoffice-base-core:amd64 (4:7.4.7-1, 4:7.4.7-1+deb12u1), 
        libreoffice-core:amd64 (4:7.4.7-1, 4:7.4.7-1+deb12u1),
        libreoffice-common:amd64 (4:7.4.7-1, 4:7.4.7-1+deb12u1), ure:amd64 (4:7.4.7-1, 4:7.4.7-1+deb12u1), 
        libreoffice-draw:amd64 (4:7.4.7-1, 4:7.4.7-1+deb12u1), i
        libreoffice-impress:amd64 (4:7.4.7-1, 4:7.4.7-1+deb12u1), 
        libreoffice-l10n-*:amd64 (4:7.4.7-1, 4:7.4.7-1+deb12u1), 
        libreoffice-style-colibre:amd64 (4:7.4.7-1, 4:7.4.7-1+deb12u1), 
        libreoffice-writer:amd64 (4:7.4.7-1, 4:7.4.7-1+deb12u1), i
        libreoffice-help-*:amd64 (4:7.4.7-1, 4:7.4.7-1+deb12u1), 
        libreoffice-help-common:amd64 (4:7.4.7-1, 4:7.4.7-1+deb12u1), 
        libreoffice-math:amd64 (4:7.4.7-1, 4:7.4.7-1+deb12u1), 
        libreoffice-gtk3:amd64 (4:7.4.7-1, 4:7.4.7-1+deb12u1), 
        libreoffice-help-en-gb:amd64 (4:7.4.7-1, 4:7.4.7-1+deb12u1), 
        libreoffice-help-en-us:amd64 (4:7.4.7-1, 4:7.4.7-1+deb12u1), 
        containerd.io:amd64 (1.6.25-1, 1.6.26-1), 
        uno-libs-private:amd64 (4:7.4.7-1, 4:7.4.7-1+deb12u1), 
        libjavascriptcoregtk-4.1-0:amd64 (2.42.2-1~deb12u1, 2.42.3-1~deb12u1), 
        linux-compiler-gcc-12-x86:amd64 (6.1.64-1, 6.1.66-1), 
        gir1.2-javascriptcoregtk-4.0:amd64 (2.42.2-1~deb12u1, 2.42.3-1~deb12u1), gir1.2-webkit2-4.0:amd64 (2.42.2-1~deb12u1, 2.42.3-1~deb12u1), 
        linux-headers-amd64:amd64 (6.1.64-1, 6.1.66-1), 
        libuno-purpenvhelpergcc3-3:amd64 (4:7.4.7-1, 4:7.4.7-1+deb12u1), libuno-cppu3:amd64 (4:7.4.7-1, 4:7.4.7-1+deb12u1), libuno-cppuhelpergcc3-3:amd64 (4:7.4.7-1, 4:7.4.7-1+deb12u1), libuno-sal3:amd64 (4:7.4.7-1, 4:7.4.7-1+deb12u1), 
        fonts-opensymbol:amd64 (4:102.12+LibO7.4.7-1, 4:102.12+LibO7.4.7-1+deb12u1), 
        linux-kbuild-6.1:amd64 (6.1.64-1, 6.1.66-1), linux-image-amd64:amd64 (6.1.64-1, 6.1.66-1), 
        base-files:amd64 (12.4+deb12u3, 12.4+deb12u4), 
        python3-uno:amd64 (4:7.4.7-1, 4:7.4.7-1+deb12u1), 
        libwebkit2gtk-4.0-37:amd64 (2.42.2-1~deb12u1, 2.42.3-1~deb12u1), 
        fsearch:amd64 (0.2.3-1+3.3, 0.2.3-1+3.4), 
        linux-libc-dev:amd64 (6.1.64-1, 6.1.66-1)

Remove: linux-headers-6.1.0-12-amd64:amd64 (6.1.52-1), linux-image-6.1.0-12-amd64:amd64 (6.1.52-1)
End-Date: 2023-12-12  07:48:34

```

**Thank YOU**
Cherrytree is one of the primary programs I use.
Thank you for writing and your support with CherryTree.

**Comment by giuspen** [comment]
Does this happen also if you run the "find in multiple nodes" from the top menu search instead of via the keyboard shortcut?
Could you try to reset your keyboard shortcuts to default in the preferences dialog and see if it happens again?
Could you reproduce this issue after running cherrytree from the terminal and report what is printed out?
Ideally could you try to rebuild from sources and run and report if still behaves that way

**Comment by keypey** [comment]
I will try to check it next weekend.

What does "reset your keyboard shortcuts to default" mean? I don't have my own shortcuts set in cherrytree.

**Comment by giuspen** [comment]
Yes basically from the preferences dialog tab keyboard shortcuts there is the possibility to reset the keyboard shortcuts to the default value. Just for testing purpose. You can save your settings before that witg File--Preferences--Export and restore later.

**Comment by keypey** [comment]
Ok.

**Comment by giuspen** [comment]
Definitely could be the same of https://github.com/giuspen/cherrytree/issues/2399 even though not clear yet, I will try and reproduce on a Debian12 VM

**Comment by keypey** [comment]
I reinstalled version 1.0.3 and ran two tests.
Logs:
```log
test1.log
-----------------------------------------------------<< >>-----

[2023-12-13 13:59:00.701] [   ] [debug] /home/keypey/.config/cherrytree/config.cfg parsed
[2023-12-13 13:59:06.928] [   ] [debug] shift images in MenuBar/context menu
[2023-12-13 13:59:14.684] [   ] [debug] Node name: index - maintanance
[2023-12-13 13:59:26.043] [   ] [debug] Search took 4 sec
[2023-12-13 13:59:26.046] [   ] [debug] Node name: sr_postgres
[2023-12-13 13:59:26.128] [   ] [debug] Node name: index - maintanance

(cherrytree:34399): Gdk-WARNING **: 13:59:26.304: Native Windows wider or taller than 32767 pixels are not supported

(cherrytree:34399): Gdk-WARNING **: 13:59:26.604: Native Windows wider or taller than 32767 pixels are not supported
[2023-12-13 13:59:26.943] [   ] [debug] Node name: *locale collate*
[2023-12-13 13:59:36.959] [   ] [debug] Node name: COLLATE

[2023-12-13 14:00:03.190] [   ] [debug] kill_callback_handler 15
```
```log
test2.log
-----------------------------------------------------<< >>-----

[2023-12-13 15:01:25.171] [   ] [debug] /home/keypey/.config/cherrytree/config.cfg parsed
[2023-12-13 15:01:28.771] [   ] [debug] shift images in MenuBar/context menu
[2023-12-13 15:01:31.668] [   ] [debug] Node name: COLLATE
[2023-12-13 15:01:39.484] [   ] [debug] Search took 3 sec
[2023-12-13 15:01:39.487] [   ] [debug] Node name: sr_postgres
[2023-12-13 15:01:39.532] [   ] [debug] Node name: COLLATE

(cherrytree:34905): Gdk-WARNING **: 15:01:39.708: Native Windows wider or taller than 32767 pixels are not supported

(cherrytree:34905): Gdk-WARNING **: 15:01:39.982: Native Windows wider or taller than 32767 pixels are not supported
[2023-12-13 15:01:39.984] [   ] [debug] Node name: *locale collate*
[2023-12-13 15:02:38.959] [   ] [debug] kill_callback_handler 15
```

In both cases, the problem described in the report occurred.
**kill_callback_handler**: force stop from text terminal.

---
I also performed an additional test on a fresh VM with D12. In this case, after the error occurred, I was logged out.

**Comment by keypey** [comment]
I work on two monitors with resolutions:
```

Screen 0: minimum 320 x 200, current 3840 x 1200, maximum 16384 x 16384
HDMI-1 connected 1920x1200+0+0 (normal left inverted right x axis y axis) 519mm x 324mm
   1920x1200     59.95*+
    ...
HDMI-2 connected 1920x1200+1920+0 (normal left inverted right x axis y axis) 519mm x 324mm
   1920x1200     59.95*+
    ...
```

**Comment by giuspen** [comment]
Many thanks @keypey I may have understood, I'll get back.

**Comment by giuspen** [comment]
Please try https://www.giuspen.net/software/cherrytree_1.0.3-2+10~Debian12_amd64.deb and report if the bug is fixed there

**Comment by keypey** [comment]
I will install it and use this version during the workday.

**Comment by keypey** [comment]
> Please try https://www.giuspen.net/software/cherrytree_1.0.3-2+10~Debian12_amd64.deb and report if the bug is fixed there

I will install it and use this version during the workday.

**Comment by keypey** [comment]
Throughout the day I used several different SQLite files, from: ~2000  to: ~6000 items. 
Works correctly.

I think you've located the problem correctly.

**Comment by giuspen** [comment]
Many thanks @keypey !

**Comment by keypey** [comment]
> Throughout the day I used several different SQLite files, from: ~2000 to: ~6000 items. Works correctly.
> 
> I think you've located the problem correctly. I'll be using it tomorrow as well - so if anything bad happens, I'll be sure to let you know.

After another two days, I can confirm with certainty that the error does not occur.

**Comment by giuspen** [comment]
Thanks @keypey I'll try and release v1.0.4 this evening, at the latest tomorrow

**Comment by keypey** [comment]
**Great!**
I'll install it as soon as it shows up!
Thank you again, such a great app and fantastic support.
--
By the way, a question:
Could the functionality be added to the search (CTRL-T): "search for icon"...

**Comment by giuspen** [comment]
I'm not sure, what do you mean with search for icon?

**Comment by keypey** [comment]
Ex.
"Search for all branches that have the following icon: ❤️ "

**Comment by giuspen** [comment]
I see, I'll keep that in mind but have some features already in the queue first

**Comment by giuspen** [comment]
Fixed in v1.0.4
